warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IsNotNullPruningVisitor.java

```diff
@@ -196,22 +298,28 @@ protected String fieldForChild(JexlNode node) {
      * @return a singular field if it exists, or null
      */
     protected String fieldForUnion(JexlNode node) {
+        String field;
         Set<String> fields = new HashSet<>(node.jjtGetNumChildren());
         for (JexlNode child : JexlNodes.children(node)) {
             child = JexlASTHelper.dereference(child);
+            field = null;
             if (child instanceof ASTEQNode || child instanceof ASTERNode) {
```

Comment:
why not a function node as well?

Reply:
Complexity and scope. I'll write the test case for this and leave it as future work if you find that agreeable.